tests/production_pause_vote_timeout.py

```diff
@@ -0,0 +1,197 @@
+walletMgr=WalletMgr(True, keepRunning=args.leave_running, keepLogs=args.keep_logs)
+
+try:
+    TestHelper.printSystemInfo("BEGIN")
+
+    cluster.setWalletMgr(walletMgr)
+
+    Print(f'producing nodes: {pnodes}, delay between nodes launch: {delay} second{"s" if delay != 1 else ""}')
+
+    # for defproducerc producing node
+    specificExtraNodeosArgs={}
+    specificExtraNodeosArgs[2]="--production-pause-vote-timeout-ms 1000"
+
+    Print("Stand up cluster")
+    # Cannot use activateIF to transition to Savanna directly as it assumes
+    # each producer node has finalizer configured.
+    if cluster.launch(pnodes=pnodes, totalNodes=totalNodes, totalProducers=pnodes, prodCount=prodCount, delay=delay, loadSystemContract=False,
+                      specificExtraNodeosArgs=specificExtraNodeosArgs,
+                      activateIF=False, signatureProviderForNonProducer=True,
+                      topo="./tests/production_pause_vote_timeout_test_shape.json") is False:
+        errorExit("Failed to stand up eos cluster.")
+
+    assert cluster.biosNode.getInfo(exitOnError=True)["head_block_producer"] != "eosio", "launch should have waited for production to change"
+
+    node0          = cluster.getNode(0)   # producer and finalizer node for defproducera
+    node1          = cluster.getNode(1)   # producer and finalizer node for defproducerb
+    producercNode  = cluster.getNode(2)   # producer node for defproducerc
+    finalizercNode = cluster.getNode(3)   # finalizer node for defproducerc
+    centerNode     = cluster.getNode(4)
+
+    Print("Set finalizer policy and start transition to Savanna")
+    # Specifically, need to configure finalizer name for finalizercNode as defproducerc
+    transId = cluster.setFinalizers(nodes=[node0, node1, finalizercNode], finalizerNames=["defproducera", "defproducerb", "defproducerc"])
+    assert transId is not None, "setfinalizers failed"
+    assert cluster.biosNode.waitForTransFinalization(transId), f"setfinalizers transaction {transId} was not rolled into a LIB block"
+    assert cluster.biosNode.waitForLibToAdvance(), "LIB did not advance after setFinalizers"
+
+    # biosNode no longer needed
+    cluster.biosNode.kill(signal.SIGTERM)
+    cluster.waitOnClusterSync(blockAdvancing=5)
+
+    Print("Wait for LIB on all producing nodes to advance")
+    assert node0.waitForLibToAdvance(), "node0 did not advance LIB"
+    assert node1.waitForLibToAdvance(), "node1 did not advance LIB"
+    assert producercNode.waitForLibToAdvance(), "producercNode did not advance LIB"
+
+    ####################### test 1 ######################
+
+    Print("Shutdown finalizercNode")
+    finalizercNode.kill(signal.SIGTERM)
+    assert not finalizercNode.verifyAlive(), "finalizercNode did not shutdown"
+
+    # wait some time for producercNode paused
+    paused = False
+    for i in range(0, 15):
+        time.sleep(1)
+        paused = not producercNode.waitForHeadToAdvance(timeout=1)
```

Comment:
Should timeout be a little higher than 1? Otherwise it could fail because of the timeout and we'd think the producer is paused.

Reply:
Thanks. Change to use producer RPC `paused` endpoint.